cirq-google/cirq_google/workflow/quantum_executable.py

```diff
@@ -245,7 +258,7 @@ def __init__(self, executables: Sequence[QuantumExecutable]):
             executables = tuple(executables)
         object.__setattr__(self, 'executables', executables)
 
-        object.__setattr__(self, '_hash', hash(dataclasses.astuple(self)))
+        object.__setattr__(self, '_hash', None)
```

Comment:
In `QuantumExecutable.__init__` we set the hash unconditionally and the comment notes that this ensures that all members are hashable at construction time. Do we want the same eager semantics here? It seems to me that we should do the same thing in both cases, either both eagerly hashed or both lazily hashed. I don't have a strong opinion one way or the other as I'm not too familiar with this code.

Reply:
I changed it back to eager hash computations for both classes.  There is a test for hashability of QuantumExecutable arguments; I agree it is better to have the same behavior in QuantumExecutableGroup.

https://github.com/quantumlib/Cirq/blob/cd3182a39bbd77c385f8443da08961ca8b2fec30/cirq-google/cirq_google/workflow/quantum_executable_test.py#L147-L148